editor/src/components/canvas/controls/classname-select.tsx

```diff
@@ -66,12 +176,164 @@ export const ClassNameSelect: React.FunctionComponent = betterReactMemo('ClassNa
+    () => ({
+      container: (styles: React.CSSProperties) => ({
+        // the outermost element. It contains the popup menu, so don't set a height on it!
+        // shouldn't contain any sizing
+        ...styles,
+        width: '100%',
+      }),
+      control: (styles) => ({
+        // need to remove styles here, since that implicitly sets a height of 38
+        // ...styles,
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+        background: 'transparent',
+        outline: 'none',
+        ':focus-within': {
+          outline: 'none',
+          border: 'none',
+        },
+      }),
+      valueContainer: () => ({
+        // the container for added options (tags) and input
+        // sibling to indicatorsContainer
+        // default styles mess with layout, so ignore them
+        display: 'flex',
+        alignItems: 'center',
+        gap: 4,
+        paddingLeft: 4,
+        paddingRight: 4,
+        paddingTop: 0,
+        paddingBottom: 0,
+        maxWidth: 0,
+      }),
+
+      multiValue: () => {
+        return {
+          cursor: 'pointer',
+          display: 'flex',
+          alignItems: 'center',
+          height: 18,
+          backgroundColor: '#191818',
+        }
+      },
+      multiValueLabel: () => ({
+        fontSize: 10,
+        padding: '2px 4px',
+        color: 'white',
+      }),
+      multiValueRemove: (styles: React.CSSProperties, { data }) => ({
+        width: 11,
+        display: 'flex',
+        paddingTop: 2,
+        opacity: 0.4,
+        color: data.color,
+        ':hover': {
+          opacity: 1,
+          backgroundColor: data.color,
+          color: 'white',
+        },
+      }),
+      input: () => {
+        return {
+          fontSize: 11,
+          color: theme.fg1.value,
+          letterSpacing: 0.3,
+          background: 'transparent',
+          display: 'flex',
+          alignItems: 'center',
+        }
+      },
+      indicatorsContainer: (styles) => ({
+        ...styles,
+        height: 20,
+      }),
+      option: (styles: React.CSSProperties, { data, isDisabled, isFocused, isSelected }) => {
+        // a single entry in the options list
+        const categories = data?.categories ?? []
+
+        let optionColor = ChromaBlack
+        if (categories.length === 1) {
+          optionColor = ChromaThemePrimary
+        }
+
+        let backgroundColor: string | undefined = undefined
+        if (isFocused) {
+          backgroundColor = optionColor.alpha(0.1).css()
+        } else if (isSelected) {
+          backgroundColor = optionColor.css()
+        } else if (isDisabled) {
+          backgroundColor = undefined
+        }
+
+        let activeBackgroundColor =
+          !isDisabled && (isSelected ? data.color : optionColor.alpha(0.3).css())
+
+        let color: string = optionColor.css()
+        if (isSelected) {
+          if (chroma.contrast(optionColor, 'white') > 2) {
```

Comment:
we could also memoize this, i'm not sure how expensive chroma.contrast() is, but I remember that chroma was not _too_ fast when we first tried it...